adafruit_imageload/png.py

```diff
@@ -13,16 +13,22 @@
 
 """
 
+try:
+    # pylint: disable=unused-import
+    import typing
+    from .displayio_types import PaletteConstructor, BitmapConstructor
+except ImportError:
+    pass
+
 import struct
 import zlib
 
-
 __version__ = "0.0.0-auto.0"
 __repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_ImageLoad.git"
 
 
 def load(
-    file, *, bitmap, palette=None
+    file: str, *, bitmap: BitmapConstructor, palette: PaletteConstructor = None
 ):  # pylint: disable=too-many-locals,too-many-branches
```

Comment:
I have been researching a little about how to annotate the return type of this correctly, without assuming that we are always going to use the `displayio.Bitmap` and `displayio.Palette`. It seems like we can do this by using type aliases (https://peps.python.org/pep-0484/#type-aliases) to make sure the function returns the same type as the constructor that it gets, and optionally maybe protocols (https://peps.python.org/pep-0544/) to make sure those types have the required methods on them.